src/firebase/index.ts

```diff
@@ -104,7 +109,18 @@ export function logInfo(code: string, payload: Record<string, any>) {
   log('INFO', code, payload);
 }
 
-// Logs an error to Stackdriver.
-export function logError(code: string, payload: Record<string, any>) {
-  log('ERROR', code, payload);
+// Logs an error or structured payload to Stackdriver.
+// The error or payload is also logged via console.error.
+export function logError(
+  code: string,
+  errorOrPayload: Error | Record<string, any>
+) {
+  if (errorOrPayload instanceof Error) {
+    const e: Error = errorOrPayload;
+    log('ERROR', code, { message: e.message, name: e.name, stack: e.stack });
+    console.error(e);
+  } else {
+    log('ERROR', code, errorOrPayload);
```

Comment:
the first parameter to log() is the severity. INFO- and ERROR-severity messages are the same otherwise: they all have an object payload. this function can take an instance of Error as a convenience so callers don't need to extract all the fields manually.

i've updated the comments to mention severity.